Modules/System Tests/Time Zone Management/src/TimeZoneManagementTest.Codeunit.al

```diff
@@ -0,0 +1,332 @@
+
+        // [THEN] Daylight saving time is not supported
+        VerifyDaylightSavingTimeIsNotSupported(SupportsDST);
+    end;
+
+    [Test]
+    procedure CheckWhetherDateTimeIsInDST()
+    var
+        DateTimeToTest: DateTime;
+        TimeZoneId: Text;
+        DateTimeIsInDST: Boolean;
+    begin
+        // [SCENARIO #0007] Check whether datetime is in DST
+        Initialize();
+
+        // [GIVEN] Time zone that has daylight saving time
+        TimeZoneId := CreateTimeZoneThatHasDaylightSavingTime();
+        // [GIVEN] DateTime inside daylight saving time period
+        DateTimeToTest := CreateDateTimeInsideDaylightSavingTimePeriod();
+
+        // [WHEN] Check whether datetime falls within DST for the time zone
+        DateTimeIsInDST := CheckWhetherDateTimeFallsWithinDSTForTheTimeZone(DateTimeToTest, TimeZoneId);
+
+        // [THEN] DateTime shows as being in DST
+        VerifyDateTimeShowsAsBeingInDST(DateTimeIsInDST);
+    end;
+
+    [Test]
+    procedure CheckWhetherDateTimeIsNotInDST()
+    var
+        DateTimeToTest: DateTime;
+        TimeZoneId: Text;
+        DateTimeIsInDST: Boolean;
+    begin
+        // [SCENARIO #0008] Check whether datetime is not in DST
+        Initialize();
+
+        // [GIVEN] Time zone that has daylight saving time
+        TimeZoneId := CreateTimeZoneThatHasDaylightSavingTime();
+        // [GIVEN] DateTime outside daylight saving time period
+        DateTimeToTest := CreateDateTimeOutsideDaylightSavingTimePeriod();
+
+        // [WHEN] Check whether datetime falls within DST for the time zone
+        DateTimeIsInDST := CheckWhetherDateTimeFallsWithinDSTForTheTimeZone(DateTimeToTest, TimeZoneId);
+
+        // [THEN] DateTime shows as not being in DST
+        VerifyDateTimeShowsAsNotBeingInDST(DateTimeIsInDST);
+    end;
+
+    [Test]
+    procedure CheckWhetherDateTimeIsInDSTForTimeZoneWithoutDST()
+    var
+        DateTimeToTest: DateTime;
+        TimeZoneId: Text;
+        DateTimeIsInDST: Boolean;
+    begin
+        // [SCENARIO #0009] Check whether datetime is in DST for time zone without DST
+        Initialize();
+
+        // [GIVEN] Time zone that does not have daylight saving time
+        TimeZoneId := CreateTimeZoneThatDoesNotHaveDaylightSavingTime();
+        // [GIVEN] DateTime inside daylight saving time period
+        DateTimeToTest := CreateDateTimeInsideDaylightSavingTimePeriod();
+
+        // [WHEN] Check whether datetime falls within DST for the time zone
+        DateTimeIsInDST := CheckWhetherDateTimeFallsWithinDSTForTheTimeZone(DateTimeToTest, TimeZoneId);
+
+        // [THEN] DateTime shows as not being in DST
+        VerifyDateTimeShowsAsNotBeingInDST(DateTimeIsInDST);
+    end;
+
+    var
+        PermissionsMock: Codeunit "Permissions Mock";
+        DateTimeOffset: Codeunit "DateTime Offset";
+        DaylightSavingTimeInfo: Codeunit "Daylight Saving Time Info";
+        LibraryAssert: Codeunit "Library Assert";
+        IsInitialized: Boolean;
```

Comment:
Let's get all of the global variables moved to the top. It's kind of a standard we have.

Reply:
Yes, you are right, of course. My test codeunit template puts them in that spot and I forgot to move them. Did that now.